src/main/java/org/spongepowered/common/mixin/core/advancement/MixinAdvancementList.java

```diff
@@ -58,8 +60,8 @@
 
     @Shadow @Final private static Logger LOGGER;
 
-    @Shadow @Final private Map<ResourceLocation, Advancement> advancements;
-    @Shadow @Final private Set<Advancement> roots;
+    @Shadow @Final private Map<ResourceLocation, Advancement> advancements = new AdvancementMap();
```

Comment:
This literally makes no sense. You're shadowing a field that is finalized in the actual class then initializing it here. Can't do that.

Reply:
As well as you'll run into mod compatibility issues with replacing the map reference that Vanilla is using..

Reply:
It's working fine like this. But I can move it if you want.

Reply:
Same is done in: https://github.com/SpongePowered/SpongeCommon/blob/df3266d7b2923018f111df3a4db442e6a4882841/src/main/java/org/spongepowered/common/mixin/core/world/gen/MixinChunkProviderServer.java#L90

Reply:
`@Mutable` is only used for setting fields after it's initialized

Reply:
The only reason why we do that for the chunk provider server is to provide better caching of the previously requested chunk, reducing lookups. If it's doing the same with the `AdvancementMap`, you should name it more appropriately.

Reply:
I am just making sure that all the Advancements are getting captured and registered using the map.